crates/sqlbuiltins/src/functions/table/object_store.rs

```diff
@@ -697,32 +678,97 @@ impl TableFunc for CloudUpload {
             storage.db_id(),
             file_name
         ));
-        let base_url = storage.root_url.to_string();
-        let base_url = ObjectStoreUrl::parse(base_url)?;
 
         // No globbing: just retrieve meta for the requested file
-        let meta: ObjectMeta = match store.head(&path).await {
+        let meta: ObjectMeta = match storage.store.head(&path).await {
             Ok(meta) => meta,
             Err(_) => {
                 return Err(ExtensionError::String(format!(
                     "file not found: {}",
-                    file_name
+                    file_name,
                 )))
             }
         };
-        let objects: Vec<ObjectMeta> = vec![meta]; // needed for ObjStoreTableProvider impl
-
-        let session_state = ctx.get_session_state();
-        let arrow_schema = file_format
-            .infer_schema(&session_state, &store.inner, &objects)
-            .await?;
-
-        Ok(Arc::new(ObjStoreTableProvider::new(
-            store.inner.clone(),
-            arrow_schema,
-            base_url,
-            objects,
-            file_format,
-        )))
+
+        let ext = Path::new(&file_name)
+            .extension()
+            .ok_or_else(|| {
+                ExtensionError::String(
+                    "missing file extension, supported: [.csv, .ndjson, .jsonl, .json, .parquet, .bson, .xlsx]".to_string(),
+                )
+            })?
+            .to_str()
+            .ok_or_else(|| {
+                ExtensionError::String(format!("unsupported file extension: {file_name}"))
+            })?
+            .to_lowercase();
+
+        match ext.as_str() {
```

Comment:
note: maybe this needs to be case normalized

Reply:
> worried that we should maybe not duplicate the code from the object store.

what code are we duplicating?